# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/KnowsCompositor.h
# Commit: 72d4bf5b093e
# Full Hash: 72d4bf5b093e9b00eed78bf4e51850d56df5ae6c
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2020-11-05 16:09:03
# Regressor Bug: 1673983
# File Overlap Count: 2
# Description:
#   Bug 1673983 - Add RenderTextureHostSWGL support to D3D11 RenderTextureHosts. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D95799
# ==============================================================================

diff -r 99b2e5c55a25 -r 72d4bf5b093e gfx/layers/ipc/KnowsCompositor.h
--- a/gfx/layers/ipc/KnowsCompositor.h	Thu Nov 05 09:14:52 2020 +0000
+++ b/gfx/layers/ipc/KnowsCompositor.h	Thu Nov 05 09:15:05 2020 +0000
@@ -103,7 +103,9 @@
   bool SupportsD3D11() const {
     return GetCompositorBackendType() == layers::LayersBackend::LAYERS_D3D11 ||
            (GetCompositorBackendType() == layers::LayersBackend::LAYERS_WR &&
-            GetCompositorUseANGLE());
+            (GetCompositorUseANGLE() ||
+             mTextureFactoryIdentifier.mWebRenderCompositor ==
+                 layers::WebRenderCompositor::D3D11));
   }
 
   bool GetCompositorUseANGLE() const {